File: exercise/kotlin/day24/src/test/kotlin/assertions/ToyAssertions.kt
package assertions

import domain.Toy
import domain.core.Event
import doubles.InMemoryToyRepository
import kotlin.test.fail

fun Toy.shouldHaveRaisedEvent(repository: InMemoryToyRepository, expectedEvent: Event) =
    repository.raisedEvents()
        .lastOrNull()
        .let { lastEvent ->
            if (lastEvent == null || lastEvent != expectedEvent) {
                fail("Raised events should contain $expectedEvent. Actual: $lastEvent")
            }
        }

File: exercise/scala/day24/src/test/scala/assertions/ToyAssertions.scala
package assertions

import domain.Toy
import domain.core.Event
import doubles.InMemoryToyRepository
import org.scalatest.Assertions.fail

object ToyAssertions {
  def shouldHaveRaisedEvent(repository: InMemoryToyRepository, expectedEvent: Event): Unit = {
    val lastEvent = repository.raisedEvents.lastOption
    if (lastEvent.isEmpty || lastEvent.get != expectedEvent) {
      fail(s"Raised events should contain $expectedEvent. Actual: $lastEvent")
    }
  }
}